.evergreen/calc_release_version_selftest.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# calc_release_version_selftest.sh is used to test output of calc_release_version.py.
+# run with:
+# cd etc
```

Comment:
Along with the comment suggesting not to run the self-test in CI, suggest moving `calc_release_version_selftest.sh` to the `etc` directory.